WooCommerce/Classes/ViewRelated/Orders/Order Creation/Synchronizer/FeesInputTransformer.swift

```diff
@@ -0,0 +1,27 @@
+import Foundation
+import Yosemite
+
+/// Helper to update an `order` given an `OrderFeeLine` input type.
+///
+struct FeesInputTransformer {
+
+    /// Adds, deletes, or updates a fee line input into an existing order.
+    ///
+    static func update(input: OrderFeeLine?, on order: Order) -> Order {
+        // If input is `nil`, then we remove any existing fee line.
+        // We remove a fee line by setting its `name` to nil.
+        guard let input = input else {
+            let linesToRemove = order.fees.map { $0.copy(name: .some(nil), total: "0") }
```

Comment:
Nit: Could we use the helper in `OrderFactory` to create the fee line to remove?

```suggestion
            let linesToRemove = order.fees.map { OrderFactory.deletedFeeLine($0) }
```

(In that case it's worth updating that helper to also set `total: "0"` like you did here, to optimistically update the order total.)

Reply:
Nice, fixed in https://github.com/woocommerce/woocommerce-ios/pull/6383/commits/6f38675572d41bafdb0e33631a83909d44dab2c3